README.md

```diff
@@ -29,23 +29,167 @@ mtx.myTranspose();
 ```
 
 7. Use `myMap` to build a new array where every element is incremented by 10.
-8. Use `myMap` to build a new array where every non-string element is converted to an empty string. 
+8. Use `myMap` to build a new array where every non-string element is converted to an empty string.
 9. Use `myFilter` to get only the even elements in an array.
-10. Use `myEvery` to check if all elements in the array are the same. 
-11. Use `myReduce` to return the sum of every element in an array. 
-12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements. 
-13. Use `myReduce` to find the largest number in an array. 
+10. Use `myEvery` to check if all elements in the array are the same.
+11. Use `myReduce` to return the sum of every element in an array.
+12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements.
+13. Use `myReduce` to find the largest number in an array.
 14. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.
 Use `myFilter` to return a new array of every element of arr that can be evenly divided by divisor.
-14. Write a `myJoin` function. 
+14. Write a `myJoin` function.
 15. Write a `mySlice` function.
 16. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
 Use reduce.
-17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index. 
+17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index.
 
-Bonus: Write a `myFlatten`. This should take a multi-dimensional array and return it as one array. 
+Bonus: Write a `myFlatten`. This should take a multi-dimensional array and return it as one array.
 ```js
 let arr = [1, 2, [3, 4, 5, [6, 7, 8]]]
   arr.myFlatten();
   // => [1, 2, 3, 4, 5, 6, 7, 8]
 ```
+//Q1
+let sampleNumArray = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+
+Array.prototype.myForEach = function (callback) {
+	for (i = 0; i < this.length; i++) {
+		callback(this[i])
+	}
+}
+
+//for each only produce side effects
+// sampleNumArray.forEach( function (name) {
+// 	console.log (name)
+// })
+
+// reed sample
+// Array.prototype.myForEach = function(callback){
+// 	for (let i = 0; i < this.length; i++){
+// 		callback(this[i])
+// 	}
+// }
+
+//ES6
+// sampleNumArray.myForEach((el) => {
+// console.log (el)
+// })
+
+// //ES5
+// sampleNumArray.myForEach( function (el){
+// 	console.log(el)
+// })
+
+
+
+//Q2
+
+Array.prototype.myMap = function(el) {
+let result = []
+this.myForEach((el) => {
+	result.push(el)
```

Comment:
so we want to pass in a callback function and call that function on each element.